pkg/util/ipvs/kernelcheck_linux.go

```diff
@@ -58,37 +58,54 @@ func (r RequiredIPVSKernelModulesAvailableCheck) Check() (warnings, errors []err
 	loadModules.Insert(mods...)
 	modules := wantModules.Difference(loadModules).UnsortedList()
 
-	// Check builtin modules exist or not
-	if len(modules) != 0 {
-		kernelVersionFile := "/proc/sys/kernel/osrelease"
-		b, err := r.Executor.Command("cut", "-f1", "-d", " ", kernelVersionFile).CombinedOutput()
-		if err != nil {
-			errors = append(errors, fmt.Errorf("error getting os release kernel version: %v(%s)", err, out))
-			return nil, errors
-		}
+	if len(modules) == 0 {
+		return nil, nil
+	}
 
-		kernelVersion := strings.TrimSpace(string(b))
-		builtinModsFilePath := fmt.Sprintf("/lib/modules/%s/modules.builtin", kernelVersion)
-		out, err := r.Executor.Command("cut", "-f1", "-d", " ", builtinModsFilePath).CombinedOutput()
+	// Check if loadable modules are installed
+	var missingModules []string
+	for _, module := range modules {
+		out, err = r.Executor.Command("modinfo", module).CombinedOutput()
 		if err != nil {
-			errors = append(errors, fmt.Errorf("error getting required builtin kernel modules: %v(%s)", err, out))
-			return nil, errors
+			warnings = append(warnings, fmt.Errorf("error getting module info for %q; error: %v; output: %s", module, err, strings.TrimSpace(string(out))))
+			missingModules = append(missingModules, module)
 		}
```

Comment:
Actually, i'm do not clear why should add this check. can you explain more details?

Reply:
sure. we run modinfo <module> to check if loadable modules are installed. If modinfo returns non-zero return code there could be different reasons for that, e.g. there is no modinfo tool in the PATH, there is no info about the module etc. It makes sense to show those reasons to users if modules are not found in my opinion. If they're found among builtin modules those warnings are ignored.